hypertension/dc-7101-algorithms/py-root/lib/utils.py

```diff
@@ -1,15 +1,145 @@
 from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import operator
 
-# TODO: This might be overwriting whatever data object is passed in as bp_readings (I dont know b/c I dont know how Python works). If this is happening, change this function so that it doesnt overwrite the original data anymore.
-def add_proper_dates(bp_readings):
-    for bp_reading in bp_readings:
-        bp_reading.update({ "date_object": datetime.fromisoformat(bp_reading["date"]) })
+def calculate_correct_bp_reading(bp_readings, use_diastolic):
+    diastolic_bucket_count = {
+        "130": 0,
+        "120": 0,
+        "110": 0,
+        "100": 0,
+        "less_than_one_hundred": 0
+    }
 
-    return bp_readings
+    systolic_bucket_count = {
+        "200": 0,
+        "160": 0,
+        "less_than_one_sixty": 0
+    }
 
+    if use_diastolic:
+        type = "diastolic"
+        bucket_count = diastolic_bucket_count
+    else:
+        type = "systolic"
+        bucket_count = systolic_bucket_count
 
-def sufficient_to_autopopulate (bp_readings):
-    # TODO: Implement algorithm
-    # BP data is sufficient to run through the predominance calculation algorithm and (thus auto-populate VBMS) if there is 1 BP reading within the month before the Date of Claim and at least 1 additional BP reading within 6 months of the Date of Claim.
+    for reading in bp_readings:
+        curr = reading[type]
 
-    return add_proper_dates(bp_readings)
+        key = None
+        if use_diastolic:
+            if curr >= 130:
+                key = "130"
+            elif curr >= 120 and curr < 130:
+                key = "120"
+            elif curr >= 110 and curr < 120:
+                key = "110"
+            elif curr >= 100 and curr < 110:
+                key = "100"
+            else:
+                key = "less_than_one_hundred"
+        else:
+            if curr >= 200:
+                key = "200"
+            elif curr >= 160:
+                key = "160"
+            else:
+                key = "less_than_one_sixty"
+
+        reading[f"{type}_key"] = key
+        bucket_count[key] += 1
```

Comment:
add a key to each reading, and count up the number of readings in each bucket

Reply:
We will refactor this in https://amida.atlassian.net/browse/MCP-1046

(Resolved)